cli/pack/systemd_dir.go

```diff
@@ -45,10 +55,109 @@ var (
 	}
 )
 
+type SystemdUnitParams struct {
+	FdLimit           *int `yaml:"fd-limit"`
+	StateboardFdLimit *int `yaml:"stateboard-fd-limit"`
+}
+
+func parseSystemdUnitParamsFile(systemdUnitParamsPath string, defaultUnitParamsPath string) (*SystemdUnitParams, error) {
+	var fileContentBytes []byte
+	var err error
+
+	if systemdUnitParamsPath == "" {
+		if _, err := os.Stat(defaultUnitParamsPath); err == nil {
+			log.Debugf("Default file with system unit params is used: %s", systemdUnitParamsPath)
+			systemdUnitParamsPath = defaultUnitParamsPath
+		} else if !os.IsNotExist(err) {
+			return nil, fmt.Errorf("Failed to use default file with system unit params: %s", err)
+		}
+	}
+
+	if systemdUnitParamsPath == "" {
+		return &SystemdUnitParams{}, nil
+	}
+
+	if _, err := os.Stat(systemdUnitParamsPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("Specified file with system unit params %s doesn't exists", systemdUnitParamsPath)
+	} else if err != nil {
+		return nil, fmt.Errorf("Impossible to use specified file %s: %s", systemdUnitParamsPath, err)
+	}
+
+	fileContentBytes, err = common.GetFileContentBytes(systemdUnitParamsPath)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read file with system unit params:  %s", err)
+	}
+
+	var systemdUnitParams SystemdUnitParams
+	if err := yaml.Unmarshal([]byte(fileContentBytes), &systemdUnitParams); err != nil {
+		return nil, fmt.Errorf("Failed to parse system unit params file %s: %s", systemdUnitParamsPath, err)
+	}
+
+	return &systemdUnitParams, nil
+}
+
+func checkMinValue(paramName string, value int, minValue int) error {
+	if value < minValue {
+		return fmt.Errorf("Incorrect value for %s: minimal value is %d", paramName, minValue)
+	}
+
+	return nil
+}
+
+func setDefaults(valuePtr **int, defaultValue int) error {
+	if valuePtr == nil {
+		return project.InternalError("Failed to set default value for fd limit parameter")
+	}
+
+	*valuePtr = new(int)
+	**valuePtr = defaultValue
+
+	return nil
+}
+
+func getSystemdUnitParams(ctx *context.Ctx) (*SystemdUnitParams, error) {
+	var err error
+
+	systemdUnitParams, err := parseSystemdUnitParamsFile(
+		ctx.Pack.SystemdUnitParamsPath,
+		filepath.Join(ctx.Project.Path, defaultSystemdUnitParamsFileName),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if systemdUnitParams.FdLimit == nil {
```

Comment:
This should be checked in `setDefaults`. So you can remove unnecessary `if else`.